### PR TITLE
Feat: add axiosDefaultConfig property

### PR DESCRIPTION
**Pull request by sundowndev**
- Add axiosDefaultConfig property
- Update types and JSDoc
- Update tests
- Update Axios package

### CODE REVIEW

File: __tests__/unit/index.ts [hunk 1]
```diff
@@ -1,23 +1,23 @@
 import * as axios from 'axios';
-
+import { AxiosStatic, AxiosError, AxiosResponse } from 'axios';
```
Comment (rimiti) [inline __tests__/unit/index.ts]: These two imports cannot be merged?

File: __tests__/unit/index.ts [hunk 2]
```diff
@@ -26,17 +26,18 @@ function failWith(status: number, message: string): axios.AxiosError {
     request: {},
     response: respondWith(status, message),
     isAxiosError: true,
+    toJSON: () => ({ status, message }),
```
Comment (rimiti) [inline __tests__/unit/index.ts]: 👍

File: __tests__/unit/index.ts [hunk 3]
```diff
@@ -66,6 +67,10 @@ const assertSuccessfulCase = async (agent: Luminator) => {
 describe('Luminator', () => {
   let agent: Luminator = new Luminator('USERNAME', 'password');
 
+  beforeEach(() => {
```
Comment (rimiti) [inline __tests__/unit/index.ts]: 👍

File: package.json [hunk 4]
```diff
@@ -6,7 +6,7 @@
   "main": "dist/index.js",
   "types": "dist/index.d.ts",
   "dependencies": {
-    "axios": "0.19.0",
+    "axios": "0.19.1",
```
Comment (rimiti) [inline package.json]: Great !

File: src/luminator.ts [hunk 5]
```diff
@@ -40,33 +46,41 @@ class Luminator {
   private proxyManagerOptions: IProxyManagerOption;
   private sessionId: number;
 
-  constructor(username: string, password: string, config: ILuminatorConfig = Luminator.DEFAULT_CONFIG) {
+  constructor(
```
Comment (rimiti) [inline src/luminator.ts]: JSDoc is missing.

File: src/luminator.ts [hunk 6]
```diff
@@ -40,33 +46,41 @@ class Luminator {
...
    */
   private static getSessionId(): number {
-    return Math.trunc(crypto.randomBytes(8).readUInt32LE(1) / 0xffffffff * 1000000);
+    return Math.trunc(
+      (crypto.randomBytes(8).readUInt32LE(1) / 0xffffffff) * 1000000,
+    );
   }
 
   /**
    * @description return status if it is a valid AxiosError
```
Comment (rimiti) [inline src/luminator.ts]: * @description return status if it is a valid AxiosError

to

 * @description Return status if it is a valid AxiosError

File: src/luminator.ts [hunk 7]
```diff
@@ -156,11 +170,14 @@ class Luminator {
...
    */
-  private getAxiosRequestConfig(params: axios.AxiosRequestConfig): axios.AxiosRequestConfig {
+  private getAxiosRequestConfig(
+    params: AxiosRequestConfig,
+  ): AxiosRequestConfig {
     return {
       timeout: Luminator.REQ_TIMEOUT,
       headers: { 'User-Agent': Luminator.USER_AGENT },
       httpsAgent: new HttpsProxyAgent(this.proxyManagerOptions),
+      ...this.axiosDefaultConfig,
```
Comment (rimiti) [inline src/luminator.ts]: 👍

File: __tests__/unit/index.ts (hunk 1 above)
Comment (sundowndev) [reply __tests__/unit/index.ts]: Nope, unfortunately

**Pull request merged**